### ISSUE TITLE
adb devices does find my phone.flutter devices does not find my phone.

### ISSUE THREAD

**Issue by CurryPaste** [flutter/flutter#149122]
### Steps to reproduce

- Install Flutter, 
- execute adb devices with devices, 
- execute Flutter devices without devices,
- I found two similar issues in the issue, but it seems that I haven't seen any solutions or reasons. [99333](https://github.com/flutter/flutter/issues/99333) and [73200](https://github.com/flutter/flutter/issues/73200)

### Expected results

![image](https://github.com/flutter/flutter/assets/51729441/136b7b39-fe34-43e6-9553-a3b50bff901d)

### Actual results

As shown in the above picture

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
PS C:\Users\Administrator> flutter devices -v
[ +188 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +7 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +49 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +6 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +36 ms] executing: G:\flutter\android-info\android-sdk\platform-tools\adb.exe devices -l
[  +92 ms] Found 3 connected devices:
[  +40 ms]   Windows (desktop) • windows • windows-x64    • Microsoft Windows [版本 10.0.19045.4412]
[   +3 ms]   Chrome (web)      • chrome  • web-javascript • Google Chrome 117.0.5938.134
[   +1 ms]   Edge (web)        • edge    • web-javascript • Microsoft Edge 125.0.2535.67
[   +7 ms] executing: G:\flutter\android-info\android-sdk\platform-tools\adb.exe devices -l
[  +72 ms]


           Run "flutter emulators" to list and start any available device emulators.

           If you expected another device to be detected, please run "flutter doctor" to diagnose potential issues. You
may also
           try increasing the time to wait for connected devices with the "--device-timeout" flag. Visit
https://flutter.dev/setup/
           for troubleshooting tips.
[  +11 ms] "flutter devices" took 372ms.
[   +5 ms] Running 0 shutdown hooks
[        ] Shutdown hooks complete
[   +5 ms] exiting with code 0
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
PS C:\Users\Administrator> flutter doctor -v
[√] Flutter (Channel stable, 3.22.0, on Microsoft Windows [版本 10.0.19045.4412], locale zh-CN)
    • Flutter version 3.22.0 on channel stable at G:\flutter\flutter_sdk_real
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (3 weeks ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3
    • Pub download mirror https://pub.flutter-io.cn

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at G:\flutter\android-info\android-sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = G:\flutter\android-info\android-sdk
    • Java binary at: G:\flutter\android-info\android-studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2023.3)
    • Android Studio at G:\flutter\android-info\android-studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = G:\flutter\android-info\android-studio
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)

[!] Proxy Configuration
    • HTTP_PROXY is set
    ! NO_PROXY is not set

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [版本 10.0.19045.4412]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 117.0.5938.134
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 125.0.2535.67
    !

[√] Network resources
    • All expected network resources are available.

```

</details>

**Comment by darshankawar** [comment]
@CurryPaste Check https://github.com/flutter/flutter/issues/73200#issuecomment-770320006 and see if it helps in your case or not.

**Comment by CurryPaste** [comment]
Thank you very much. I have solved this problem.😄
I deleted all environment variables and related programs and reinstalled them.
It can only be said that the installation experience of the new version is too good👍. There is no need to consider Java environment variables. A long time ago, when writing Flutter, I needed to manually configure Java environment variables. I deleted the old environment variables and reinstalled them to detect them properly.
